src/main/java/com/microsoft/azure/functions/worker/chain/SdkTypeMiddleware.java

```diff
@@ -97,4 +91,43 @@ public void invoke(MiddlewareContext context, MiddlewareChain chain) throws Exce
 
         chain.doNext(context);
     }
+
+    /**
+     * Populate the {@link SdkTypeMetaData} with required field values and ensure
+     * every required key is present. Throws a single informative exception if any
+     * are missing. 
+     * 
+     * After successfully filling in all required data, a call to 
+     * SdkTypeMetaData::parseAndVerify is made using the input metaData.
+     */
+    private void fillMetaData(ExecutionContextDataSource execCtx, SdkTypeMetaData metaData, String inputSourceName) {
+
+        BindingDataStore dataStore = execCtx.getDataStore();
+        Set<String> requiredKeys = metaData.getRequiredFields();
+        List<String> missing = new ArrayList<>();
+
+        for (String key : requiredKeys) {
+            Optional<BindingData> opt = dataStore
+                    .getDataByNameFromInputSource(key, String.class, inputSourceName);
+
+            if (opt.isPresent()) {
+                metaData.setFieldValue(key, opt.get().getValue());
+            } else {
+                missing.add(key);
+            }
+        }
+
+        if (missing.isEmpty()) {
```

Comment:
Why you need metaData.parseAndVerify() looks unecessary to me.

simply do as following 
```
if(!missing.isEmpty()){
        String message = String.format(
            "Parameter '%s' in function '%s' (invocation %s) is missing %d required key(s): %s",
            inputSourceName,
            execCtx.getFunctionName(),
            execCtx.getInvocationId(),
            missingKeys.size(),
            String.join(", ", missingKeys)
        );
        throw new IllegalArgumentException(message);
        
        ```